src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -14,12 +14,12 @@ public partial class DeflateStream : Stream
     {
         private const int DefaultBufferSize = 8192;
 
-        private Stream _stream;
+        private Stream _stream = null!; // fields initialized in init methods called from constructor
         private CompressionMode _mode;
         private bool _leaveOpen;
-        private Inflater _inflater;
-        private Deflater _deflater;
-        private byte[] _buffer;
+        private Inflater _inflater = null!;
+        private Deflater _deflater = null!;
+        private byte[] _buffer = null!;
```

Comment:
Seems like these should all be nullable.  Let's avoid too much reliance on `= null!`... it ends up largely defeating the purpose of the feature.

Reply:
(It looks like the initialization methods that are called from the ctor are only called from that one place... they can be manually inlined rather than remaining as separate methods.)

Reply:
> Seems like these should all be nullable. Let's avoid too much reliance on = null!... it ends up largely defeating the purpose of the feature.

Will update

> (It looks like the initialization methods that are called from the ctor are only called from that one place... they can be manually inlined rather than remaining as separate methods.)

On of the inits used once, moved that, but leaving other which used in 2 places